lib/defopt.py

```diff
@@ -699,7 +703,7 @@ def _get_type_from_doc(name, globalns):
     # Support for sphinx-specific "list[type]", "tuple[type]" syntax; only
     # needed for Py<3.9.
     # (This intentionally won't catch `List` or `typing.List`.)
-    match = re.match(r'(list|tuple)\[([\w\.]+)\]', name)
+    match = re.match(r'(list|tuple)\[([\w\.,]+)\]', name)
```

Comment:
the comma is specifically for the case where you have a multiple-item tuple + union operator typed in a docstring rather than a type hint